core/server/api/webhooks.js

```diff
@@ -0,0 +1,151 @@
+// # Webhooks API
+// RESTful API for creating webhooks
+// also known as "REST Hooks", see http://resthooks.org
+var Promise = require('bluebird'),
+    _ = require('lodash'),
+    https = require('https'),
+    url = require('url'),
+    pipeline = require('../utils/pipeline'),
+    apiUtils = require('./utils'),
+    models = require('../models'),
+    errors = require('../errors'),
+    logging = require('../logging'),
+    i18n = require('../i18n'),
+    docName = 'webhooks',
+    webhooks;
+
+// TODO: Use the request util. Do we want retries here?
+function makeRequest(webhook, payload) {
+    var reqOptions, reqPayload, req;
+
+    reqOptions = url.parse(webhook.get('target_url'));
+    reqOptions.method = 'POST';
+    reqOptions.headers = {'Content-Type': 'application/json'};
+
+    reqPayload = JSON.stringify(payload);
+
+    logging.info('webhook.trigger', webhook.get('event'), webhook.get('target_url'));
+    req = https.request(reqOptions);
+
+    req.write(reqPayload);
+    req.on('error', function (err) {
+        // when a webhook responds with a 410 Gone response we should remove the hook
+        if (err.status === 410) {
+            logging.info('webhook.destroy (410 response)', webhook.get('event'), webhook.get('target_url'));
+            return models.Webhook.destroy({id: webhook.get('id')});
+        }
+
+        // TODO: use i18n?
+        logging.error(new errors.GhostError({
+            err: err,
+            context: {
+                id: webhook.get('id'),
+                event: webhook.get('event'),
+                target_url: webhook.get('target_url')
+            }
+        }));
+    });
+    req.end();
+}
+
+function makeRequests(webhooksCollection, payload) {
+    _.each(webhooksCollection.models, function (webhook) {
+        makeRequest(webhook, payload);
+    });
+}
+
+/**
+ * ## Webhook API Methods
+ *
+ * **See:** [API Methods](index.js.html#api%20methods)
+ */
+webhooks = {
+
+    /**
+     * ### Add
+     * @param {Webhook} object the webhook to create
+     * @returns {Promise(Webhook)} newly created Webhook
+     */
+    add: function add(object, options) {
+        var tasks;
+
+        /**
+         * ### Model Query
+         * Make the call to the Model layer
+         * @param {Object} options
+         * @returns {Object} options
+         */
+        function doQuery(options) {
+            return models.Webhook.getByEventAndTarget(options.data.webhooks[0].event, options.data.webhooks[0].target_url, _.omit(options, ['data']))
```

Comment:
how about a composite constraint? 
event+target_url are unique.

plus you can add lowercase validation to the schema for the event column.

Reply:
> how about a composite constraint?

Sure, where/how would I add that? 😁 I couldn't see any examples of composite constraints in our schema.

> plus you can add lowercase validation to the schema for the event column.

👍 do we have an existing validation for that or do I need to add one?

Reply:
>  I couldn't see any examples of composite constraints in our schema.

Surprise, there is none 😬 

I think this could work? Have not tested it.

We could add a composite file [here](https://github.com/TryGhost/Ghost/tree/master/core/server/data/schema), which e.g. exports this:

`exports webhooks: { eventurl: ['target_url', 'event'] }`

and ensure we add these constraints after column creation [here](https://github.com/TryGhost/Ghost/blob/master/core/server/data/schema/commands.js#L88).

>  do we have an existing validation for that or do I need to add one?

We are using the `validator` package, which has `isLowerCase`.

Reply:
For ref: https://github.com/chriso/validator.js